modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/app/service/test/DLAppServiceWhenUpdatingAFileEntryTest.java

```diff
@@ -488,4 +488,77 @@ public void testShouldSucceedWithNullInputStream() throws Exception {
 			serviceContext);
 	}
 
+	@Test
+	public void testUpdateFileEntryPublishingAFileShouldOnlyHaveOneDraft()
+		throws Exception {
+
+		_testUpdateFileEntryShouldOnlyHaveOneDraft(null);
+	}
+
+	@Test
+	public void testUpdateFileEntrySchedulingAFileShouldOnlyHaveOneDraft()
+		throws Exception {
+
+		Date displayDate = new Date(System.currentTimeMillis() + Time.DAY);
+
+		_testUpdateFileEntryShouldOnlyHaveOneDraft(displayDate);
+	}
+
+	private void _testUpdateFileEntryShouldOnlyHaveOneDraft(Date displayDate)
```

Comment:
Maybe we could rename this to something similar to 

`testUpdateFileEntryShouldOnlyHaveOneDraAssetEntryforDrafts` (or something similar) in order to improve the semantics (because we are counting assetEntries in the assert)

Reply:
renamed